documentation/MIGRATION-guide-for-modularized-libraries.md

```diff
@@ -0,0 +1,253 @@
+
+## Why switching to the modularized libraries?
+
+We recommend reviewing the [complete guide for the modularized libraries](https://devblogs.microsoft.com/azure-sdk/azure-sdk-modularized-libraries-for-javascript/) for full details. Compared to the traditional libraries, Modular SDKs have following key benefits:
+
+1. Subpath exports: Modular SDKs use [subpath exports](https://nodejs.org/api/packages.html#subpath-exports)(available since Node.js version 12.7) to offer layered APIs. In which service client layer from `.` root subpath would provide similar experience to traditional client. And the API layer from `./api` subpath would provide more lightweight client context for shared state across operations.
+1. Bundle size optimization: Modular SDKs leverage @azure-rest/core-client, which offers improved bundle size efficiency compared to the previous Azure core libraries. This core package provides a general-purpose REST client, while each service-specific package includes its own TypeScript type definitions. These TypeScript types are excluded from the final asset bundle, helping to minimize overall bundle size.
+1. Long-running operations: Instead of two methods (beginDoSth and beginDoSthAndWait) in the traditional clients for each long-running operation, which are both redundant and confusing to customers. Modular SDKs offer a single method (doSth) that supports both async and sync usage.
+
+
+## How to migrate to the modularized libraries?
+
+If you're updating an existing app to use Modular SDKs, focus on these areas:
+
+1. Long-running Operations
+1. List Operations
+1. Model Property Flattening
+
+### Long-running Operations
+
+Many operations may take a long time to finish before receiving the desired response named long-running operations. We re-designed LRO in Modular SDKs. The changes mainly are three parts:
+
+- Method signature changes
+- LRO poller changes from SimplePollerLike to PollerLike
+- Rehydration changes
+
+#### Method signature changes
+
+Taking a simple LRO operation as an example with operationId `IntegrationRuntimes_Start`. In traditional client we would have two methods([link](https://github.com/Azure/azure-sdk-for-js/blob/8c1c0027d79354d2b91b318c4ceb52e462f7db92/sdk/datafactory/arm-datafactory/src/operationsInterfaces/integrationRuntimes.ts#L193)).
+
+```ts
+beginStart(
+    options?: IntegrationRuntimesStartOptionalParams,
+  ): Promise<
+    SimplePollerLike<
+      OperationState<IntegrationRuntimesStartResponse>,
+      IntegrationRuntimesStartResponse
+    >
+  >;
+beginStartAndWait(
+    options?: IntegrationRuntimesStartOptionalParams,
+  ): Promise<IntegrationRuntimesStartResponse>;
+```
+Now we would only have one method in Modular SDK.
+
```

Comment:
```suggestion
**Migrate your usage**
```ts
// Before (AutoRest-generated)
const result = await beginStartAndWait();

const poller = await beginStart();
const result2 = await poller.pollUntilDone();

// After (TypeSpec-generated)
const result = await start();           // awaiting returns the final result
const poller = start();                 // direct access to the poller
await poller.submitted();               // optional: await initial submission
const result2 = await poller;           // or: await poller.pollUntilDone()
```
```